main.go

```diff
@@ -224,49 +227,149 @@ func main() {
 
 func prepareFilter(filters []string) (tracee.Filter, error) {
 
-	uids := []uint32{}
-
 	filterHelp := "\n--filter allows you to specify values to match on for fields of traced events.\n"
 	filterHelp += "The following options are currently supported:\n"
 	filterHelp += "uid: only trace processes or containers with specified uid(s).\n"
 	filterHelp += "\t--filter uid=0                                                | only trace events from uid 0\n"
 	filterHelp += "\t--filter uid=0,1000                                           | only trace events from uid 0 or uid 1000\n"
 	filterHelp += "\t--filter uid=0 --filter uid=1000                              | only trace events from uid 0 or uid 1000 (same as above)\n"
+	filterHelp += "\t--filter 'uid>0'                                              | only trace events from uids greater than 0"
+	filterHelp += "\t--filter 'uid>0' --filter 'uid<1000'                          | only trace events from uids between 0 and 1000"
+	filterHelp += "\t--filter 'uid>0' --filter uid!=1000                           | only trace events from uids greater than 0 but not 1000"
 
 	if len(filters) == 1 && filters[0] == "help" {
 		return tracee.Filter{}, fmt.Errorf(filterHelp)
 	}
 
+	validFilterOptions := []string{"uid"}
+
+	filter := tracee.Filter{
+		UIDFilter: baseUIDFilter(),
+	}
+
 	for _, f := range filters {
-		s := strings.Split(f, "=")
-		if len(s) != 2 {
-			return tracee.Filter{}, fmt.Errorf(filterHelp)
+
+		// Parse the filter option (i.e. uid, ...)
+		isValid := false
+		filterSlice := []string{}
+		for i := range validFilterOptions {
+			if strings.HasPrefix(f, validFilterOptions[i]) {
+				filterSlice = strings.SplitAfter(f, validFilterOptions[i])
+				isValid = true
+			}
 		}
-		if !validFilterOption(s[0]) {
-			return tracee.Filter{}, fmt.Errorf("invalid filter: %s\n%s", s[0], filterHelp)
+		if !isValid {
+			return tracee.Filter{}, fmt.Errorf("invalid filter option specified, use '--filter help' for more info")
 		}
 
-		if s[0] == "uid" {
-			values := strings.Split(s[1], ",")
-			for _, v := range values {
-				uid, err := strconv.ParseUint(v, 10, 32)
-				if err != nil {
-					return tracee.Filter{}, fmt.Errorf("specified invalid uid: %s", v)
-				}
-				uids = append(uids, uint32(uid))
+		if filterSlice[0] == "uid" {
+			err := parseUIDFilter(filterSlice[1], &filter.UIDFilter)
+			if err != nil {
+				return tracee.Filter{}, err
 			}
 		}
 	}
-	return tracee.Filter{
-		UIDs: uids,
-	}, nil
+
+	appendDefaultUIDs(&filter.UIDFilter)
+
+	return filter, nil
+}
+
+// Operator represents a comparison operator such as '=', '!=', '>', or '<'
+type Operator uint8
+
+const (
+	opInvalid Operator = iota
+	opEqual
+	opNotEqual
+	opGreater
+	opLess
+)
+
+func parseUIDFilter(operatorAndValues string, uidFilter *tracee.UIDFilter) error {
+	validFilterOperators := []string{"=", "!=", ">", "<"}
+	isValid := false
+	valueSlice := []string{}
+	for i := range validFilterOperators {
```

Comment:
You can remove this loop and above variables as you already check for this with parseOperator() below.
You can give it the first character of operatorAndValues as an argument  instead of slicing here

Reply:
+1

Reply:
And this one?